### PR TITLE
chore(gradle): Mac package distribution with universal launcher and bundle java runtime for  both x64 and arm

### PR DESCRIPTION
**Pull request by miurahr**
## Pull request type

- Build and release changes -> [build/release]

## Which ticket is resolved?

May help

- Option-arrow not working in arm macOS
   *  https://sourceforge.net/p/omegat/bugs/1197/

## What does this PR change?

-  Configure tasks armMacDistZip armMacNotarize armMacSignedDistZip and armMacStapledNotarizedDistZip
-  default target "mac" includes armMac distribution

## Other information

Depends on #679

### CODE REVIEW

**Review by brandelune — COMMENTED** [review]
I only have an intel machine. I can't test.

**Review by kosivantsov — CHANGES_REQUESTED** [review]
It would be good if both zip distributions for Mac were clearly marked as to what architecture they are intended for:
OmegaT_x.x.x_Mac_x64.zip
OmegaT_x.x.x_Mac_arm.zip

**Comment by miurahr** [comment]
> I only have an intel machine. I can't test.

Now we generate universal launcher binary `OmegaT.app/Contents/MacOS/OmegaT` and resources when build time.
@brandelune  could you test it to check a degrade.

@kosivantsov I've adjusted a suffix reflect your comment.

File: build.gradle [hunk 1]
```diff
@@ -606,118 +672,133 @@ task hunspellJarSignedContents(type: Sync) {
...
-        }
-    }
-}
 
 macDistZip {
     onlyIf {
         condition(!macJRE.empty, 'JRE not found')
     }
-    archiveFileName.set("${applicationName}_${project.version}${omtVersion.beta}_Mac.zip")
+    archiveFileName.set("${applicationName}_${project.version}${omtVersion.beta}_Mac_x86.zip")
```
Comment (miurahr) [inline build.gradle:685]:
```suggestion
    archiveFileName.set("${applicationName}_${project.version}${omtVersion.beta}_Mac_x64.zip")
```

Here is a typo.

**Comment by miurahr** [comment]
Most of `OmegaT.app` bundle files are generated into `build/appbundler` and pack into `..._Mac_x64.zip` and `..._Mac_arm.zip` distribution with proper JRE.

@brandelune  Please place Java 11 runtime for x64 mac on a directory where OmegaT repository directory exists.
https://adoptium.net/download/

For test, please run `./gradlew mac`

**Comment by miurahr** [comment]
Commit d4baa7eba816abbcfd0bd60e57a9b2c699a46c83 come from #679

**Comment by miurahr** [comment]
Now launcher become an universal binary

**Comment by miurahr** [comment]
There are artifacts which are built with proposed changes.

- [OmegaT_6.1.0_Beta_Mac_arm.zip](https://omegat.sourceforege.io/temporary/OmegaT_6.1.0_Beta_Mac_arm.zip) 345M
- [OmegaT_6.1.0_Beta_Mac_x64.zip](https://omegat.sourceforege.io/temporary/OmegaT_6.1.0_Beta_Mac_x64.zip) 346M

**Comment by miurahr** [comment]
@brandelune  could you check x64 version not to break Icons, messages and execution on Intel Mac?

@kosivantsov  could you check arm version?

**Comment by brandelune** [comment]
The links don't seem to work.

**Comment by brandelune** [comment]
Proper link is here:
https://omegat.sourceforge.io/temporary/OmegaT_6.1.0_Beta_Mac_x64.zip
https://omegat.sourceforge.io/temporary/OmegaT_6.1.0_Beta_Mac_arm.zip

But there is an error message:

`Files on SourceForge.net Project Web sites are limited to a max size. The file requested is too large. If you are the project administrator, you should host your large files using the File Release Service, not Project Web. Contact [SourceForge.net support](http://sourceforge.net/support/) for assistance.`

**Comment by miurahr** [comment]
I'd like to remove files on project web because sourceforge don't support it.

**Comment by miurahr** [comment]
@brandelune could you suggest where I can put  a 350MB package for you?

**Comment by brandelune** [comment]
> @brandelune could you suggest where I can put a 350MB package for you?

Do you have a private FTP site somewhere?

File: build.gradle [hunk 2]
```diff
@@ -426,6 +431,39 @@ task webManual(type: Sync) {
...
+                identifier: 'org.omegat.OmegaT',
+                icon: 'images/OmegaT.icns',
+                version: '${version}',
+                jvmrequired: javaVersion,
+                shortversion: '${version}',
+                mainclassname: mainClassName) {
+            option(value: "-Xdock:name=${applicationName}")
+            option(value: "-Dapple.awt.application.name=${applicationName}")
+            option(value: "-Dapple.awt.application.appearance=system")
+            argument(value: '--config-file=${configfile}')
```
Comment (miurahr) [inline build.gradle:454]: It seems not working. There is no definition "configfile" and it does not replace with variable because string has single quote.

**Comment by miurahr** [comment]
> Do you have a private FTP site somewhere?

no. you can build it by yourself.

**Comment by miurahr** [comment]
arm https://drive.proton.me/urls/DP07WEFG98#Ns0qqjZLIl7l
intel https://drive.proton.me/urls/YGXYH1398M#TBE3FGroCWzH

**Comment by brandelune** [comment]
> arm https://drive.proton.me/urls/DP07WEFG98#Ns0qqjZLIl7l
> intel https://drive.proton.me/urls/YGXYH1398M#TBE3FGroCWzH

The intel version seems to work well.

**Comment by miurahr** [comment]
when intel version working, there is no defeat. We can provide new value for users.
If no objection, I'd like to ask users feedback on weekly version.

**Comment by kosivantsov** [comment]
> arm https://drive.proton.me/urls/DP07WEFG98#Ns0qqjZLIl7l
> intel https://drive.proton.me/urls/YGXYH1398M#TBE3FGroCWzH

Arm version works well too.

**Comment by brandelune** [comment]
The only (small) issue I have with this packaging is the location/URI of the manual pages when they are displayed in the browser.

**Comment by miurahr** [comment]
> The only (small) issue I have with this packaging is the location/URI of the manual pages when they are displayed in the browser.

Is the thing specific to the packaging? Is a  behavior observed in distribution without JRE?

**Comment by brandelune** [comment]
The path with the packaging is:

file:///private/var/folders/q2/qvrspt8x5r7gqj7_3_b3hqm80000gq/T/AppTranslocation/9F41BDB0-5EE4-4292-85F5-52517EE2C97C/d/OmegaT.app/Contents/Java/docs/en/index.html

Without the packaging it is:
file:///Users/suzume/Downloads/OmegaT_6.1.0_Beta_Mac_x64/OmegaT.app/Contents/Java/docs/en/index.html

In both cases, the binary is taken from the Downloads directory.

**Comment by miurahr** [comment]
There is an idea that contents can be extracted in  `docs/<lang>/` on application at first time,  and reuse it  in second time.
I wandering OmegaT has a permission to write into application folder.

**Pull request merged**